pytest_fixtures/component/os.py

```diff
@@ -17,11 +17,20 @@ def default_os(
     """
     os = getattr(request, 'param', None)
     if os is None:
-        search_string = 'name="RedHat" AND (major="6" OR major="7" OR major="8")'
+        from robottelo.config import settings
+
+        default_name = settings.server.default_os.name
+        default_major_versions = " OR ".join(
+            [f'major="{v}"' for v in settings.server.default_os.major_versions]
+        )
+        search_string = f'name="{default_name}" AND ({default_major_versions})'
     else:
-        version = os.split(' ')[1].split('.')
-        search_string = f'family="Redhat" AND major="{version[0]}" AND minor="{version[1]}"'
-    os = entities.OperatingSystem().search(query={'search': search_string})[0].read()
+        name, version = os.split(' ')
+        version = version.split('.')
+        search_string = f'name="{name}" AND major="{version[0]}" AND minor="{version[1]}"'
+    os = (
+        entities.OperatingSystem().search(query={'search': search_string, 'per_page': 50})[0].read()
```

Comment:
AFAIK the default is 20 or 25 results. If `per_page` is omitted it can happen (happened on my local instance) that not all results are displayed (depending on how many operating systems have been created by robottelo in the test run so far). I'm not even sure if 50 is enough. Is there maybe a setting that returns 'all' results?

Reply:
@dosas you are right, however, this method takes whatever is returned and only picks the 1st element, so `per_page` 20 or 50 should not make any difference.